### DISCUSSION TITLE
fail2ban based on email content with Rspamd or Sieve filters?

### DISCUSSION THREAD

**Discussion by TonyGravagno** [fail2ban/fail2ban#3280] [Q&A] [answered]
If we determine that we are receiving a lot of spam from an email server that otherwise seems to be legitimate, there May be justification to block the sending server. Of course there are a lot of nuances to this. But consider a stream of spam that might occur over a period of a few days or weeks, until the server admin is aware of the issue and hopefully deals with it.

I know there is a sieve.conf that traps authentication failures on sieved. I'm thinking about basic email filtering which results in a positive Spam vs Ham determination, and then logging the source of the spam-permitting mail servers.

In this case I will gladly accept "hey dummy, you forgot about 'this' feature", and deal with a little embarrassment. But at the moment I can't think of the "right" or "common" way to do this.

Thanks!

**Comment by sebres** [comment, marked as answer]
Just few cents from my side:
- it is possible to ban mails instead of IP (so you wouldn't need to ban the whole server) - https://github.com/fail2ban/fail2ban/wiki/How-to-ban-something-other-as-host-(IP-address),-like-user-or-mail,-etc.
Of course if you can write a custom action adding banned mail address to some blacklist of the mail server in order to reject them.
- `bantime.increment = true` may help to ban recidive (no matter IP or mail address based banning)… so one can start with small initial `bantime` in order to minimize the risk for possible false positives to ban them for long time
- the filter or just a failregex may be written for log of spam detection software if it could write some entry by reaching of some score; alternatively one can inform fail2ban about a failure directly using pickle protocol (python) or with command `fail2ban-client set <jail> attempt <id> [<failure-message1> ... <failure-messageN>]` (see #2351)

As for IP-based banning - I would not recommend to do that for several reasons, starting from the fact that single spammer sending many mails can cause a block for the whole SMTP server so could affect a couple of legitimate users of same domain (think about @gmail.com and co) and ending with impossibility to do that if the mail can be sent from several SMTP hosts, if SPF/DKIM/DMARC allowing that (again, the mails of @gmail.com can be sent from a lot of addresses)
